src/Lithnet.GoogleApps.MA/ApiInterfaces/ApiInterfaceCourseTeachers.cs

```diff
@@ -0,0 +1,347 @@
+                        {
+                            existingChange.ValueChanges.Add(valueChange);
+                        }
+                    }
+                    else
+                    {
+                        adds = changes.Where(t => t.ModificationType == ValueModificationType.Add).Select(t => t.Value).ToList();
+
+                        if (adds.Count > 0)
+                        {
+                            attributeChanges.Add(AttributeChange.CreateAttributeAdd(attributeName, adds));
+                        }
+                    }
+                    break;
+
+                case AttributeModificationType.Replace:
+                    if (existingChange != null)
+                    {
+                        attributeChanges.Remove(existingChange);
+                    }
+
+                    adds = changes.Where(t => t.ModificationType == ValueModificationType.Add).Select(t => t.Value).ToList();
+                    if (adds.Count > 0)
+                    {
+                        attributeChanges.Add(AttributeChange.CreateAttributeReplace(attributeName, adds));
+                    }
+
+                    break;
+
+                case AttributeModificationType.Update:
+                    if (existingChange != null)
+                    {
+                        if (existingChange.ModificationType != AttributeModificationType.Update)
+                        {
+                            throw new InvalidOperationException();
+                        }
+
+                        foreach (ValueChange valueChange in changes)
+                        {
+                            existingChange.ValueChanges.Add(valueChange);
+                        }
+                    }
+                    else
+                    {
+                        if (changes.Count > 0)
+                        {
+                            attributeChanges.Add(AttributeChange.CreateAttributeUpdate(attributeName, changes));
+                        }
+                    }
+
+                    break;
+
+                case AttributeModificationType.Delete:
+                case AttributeModificationType.Unconfigured:
+                default:
+                    throw new InvalidOperationException();
+            }
+        }
+
+
+        private void GetTeacherChangesFromCSEntryChange(CSEntryChange csentry, out CourseTeachers adds, out CourseTeachers deletes, out CourseTeachers reportedAdds, out CourseTeachers reportedDeletes, bool replacing)
+        {
+            adds = new CourseTeachers();
+            deletes = new CourseTeachers();
+            reportedAdds = new CourseTeachers();
+            reportedDeletes = new CourseTeachers();
+
+            CourseTeachers existingTeachers;
+
+
+            if (replacing)
+            {
+                // TODO: Get existing teachers? Is it needed?
+                existingTeachers = new CourseTeachers();
```

Comment:
If FIM sends an attribute replace, rather than an update, then you'll  end up with an exported-change-not-reimported error if you don't work the deletes our yourself.

Reply:
Thanks. I'll add code to fetch the Teachers (and Students) when replacing.